clar2wasm/tests/standard/unit_tests.rs

```diff
@@ -1826,3 +1826,297 @@ fn bit_not_int() {
+        &mut store,
+        &[Val::I64(0), Val::I64(0), Val::I64(0), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 1);
+    assert_eq!(result[1].unwrap_i64(), 0);
+
+    // pow(1, 0) == 1
+    pow.call(
+        &mut store,
+        &[Val::I64(1), Val::I64(0), Val::I64(0), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 1);
+    assert_eq!(result[1].unwrap_i64(), 0);
+
+    // pow(2, 0) == 1
+    pow.call(
+        &mut store,
+        &[Val::I64(2), Val::I64(0), Val::I64(0), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 1);
+    assert_eq!(result[1].unwrap_i64(), 0);
+
+    // pow(0, 1) == 0
+    pow.call(
+        &mut store,
+        &[Val::I64(0), Val::I64(0), Val::I64(1), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 0);
+    assert_eq!(result[1].unwrap_i64(), 0);
+
+    // pow(123, 1) == 123
+    pow.call(
+        &mut store,
+        &[Val::I64(123), Val::I64(0), Val::I64(1), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 123);
+    assert_eq!(result[1].unwrap_i64(), 0);
+
+    // pow(3, 2) == 9
+    pow.call(
+        &mut store,
+        &[Val::I64(3), Val::I64(0), Val::I64(2), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 9);
+    assert_eq!(result[1].unwrap_i64(), 0);
+
+    // pow(3, 3) == 27
+    pow.call(
+        &mut store,
+        &[Val::I64(3), Val::I64(0), Val::I64(3), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 27);
+    assert_eq!(result[1].unwrap_i64(), 0);
+
+    // pow(3, 80) = large number
+    pow.call(
+        &mut store,
+        &[Val::I64(3), Val::I64(0), Val::I64(80), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 4389419161382147137);
+    assert_eq!(result[1].unwrap_i64(), 8012732698178659004);
+
+    // pow(3, 81) overflows
+    pow.call(
+        &mut store,
+        &[Val::I64(3), Val::I64(0), Val::I64(81), Val::I64(0)],
+        &mut result,
+    )
+    .expect_err("expected overflow");
+
+    // pow(2, 126) = 1 << 126
+    pow.call(
+        &mut store,
+        &[Val::I64(2), Val::I64(0), Val::I64(126), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-uint failed");
+    assert_eq!(result[0].unwrap_i64(), 0);
+    assert_eq!(result[1].unwrap_i64(), 0x4000000000000000u64 as i64);
+
+    // pow(2, 127) overflows
+    pow.call(
+        &mut store,
+        &[Val::I64(2), Val::I64(0), Val::I64(127), Val::I64(0)],
+        &mut result,
+    )
+    .expect_err("expected overflow");
```

Comment:
Can you add `(pow -2 127)`?

Reply:
Great catch! I didn't think about that edge case!

Reply:
I added a shortcut for `pow(2, x)`, it deals with this edge case.

See 991aabbe819ec2dc6085d5c679cae240c578f712